File: Leet_150/217_contains_duplicate/DuplicateChecker.py
class DuplicateChecker(object):
    def containsDuplicate(nums):
        nums.sort()  # Sort the array
    # Check adjacent elements for duplicates
        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                return True  # Duplicate found
        return False  # No duplicate found

def main():
    # Getting number of elements from user
    n = int(input("Enter the number of elements: "))
    # Storing the array elements from user
    nums = []
    print("INPUT:")
    for _ in range(n):
        nums.append(int(input()))
    
    has_duplicate = containsDuplicate(nums)

    # Print the result
    if has_duplicate:
        print("OUTPUT: true")
    else:
        print("OUTPUT: false")

if __name__ == "__main__":
    main()

File: Leet_150/217_contains_duplicate/DuplicateChecker.java
import java.util.Arrays;
import java.util.Scanner;

public class DuplicateChecker {
    public static boolean containsDuplicate(int[] num) {
        Arrays.sort(num); // this sorts the array
        // check adjacent elements for duplicate
        for (int i = 0; i < num.length - 1; i++) {
            if (num[i] == num[i + 1]) {
                return true;// duplicate found
            }
        }
        return false;// No ducplicate found

    }

    public static void main(String[] args) {
        // getting no of elements from user
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of elements:");
        int n = sc.nextInt();
        // storing the array elements from user
        int num[] = new int[n];
        System.out.print("INPUT: ");
        for (int i = 0; i < n; i++) {
            num[i] = sc.nextInt();
        }

        boolean hasDuplicate = containsDuplicate(num);

        // Print the result
        if (hasDuplicate) {
            System.out.print("OUTPUT: true");
        } else {
            System.out.print("OUTPUT: false");
        }
        sc.close();
    }
}